server/pairing/statecontrol/state_management.go

```diff
@@ -19,6 +19,9 @@ type ProcessStateManager struct {
 	pairing     bool
 	pairingLock sync.Mutex
 
+	messageSyncEnabled bool
+	messageSyncLock    sync.Mutex
```

Comment:
Not sure why we need any syncing here.
But `atomic.Bool` is a better option anyway